### PR TITLE
ci(jenkins): trigger async ITs and update github check for the pull request

### PR DESCRIPTION
**Pull request by v1v**
fixes https://github.com/elastic/apm-integration-testing/issues/41

## Highlights
- It's triggered for each PR by default and disabled to any upstream branches.
- Trigger the Integration Tests, aka ITs, pipelines from https://github.com/elastic/apm-integration-testing based on PRs or on demand, aka manually.
- It does trigger only the `PHP` stage in the ITs.
- ITs should be async.
- This new stage notifies the GitHub check with the status of the execution.
- It runs immediately after the build/test stage and before the package-generation stage.

## Tasks
- [x] It does require to support ITs for PHP

## Issues

Caused by https://github.com/elastic/apm-integration-testing/pull/863 and https://github.com/elastic/apm-integration-testing/issues/865

## Tests

- Downstream -> [build](https://apm-ci.elastic.co/job/apm-integration-tests-selector-mbp/view/change-requests/job/PR-946/2/)

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -275,6 +277,22 @@ pipeline {
...
+          agent none
+          when {
+            beforeAgent true
+            changeRequest()
+          }
+          steps {
+            githubNotify(context: "${env.GITHUB_CHECK_ITS_NAME}", description: "${env.GITHUB_CHECK_ITS_NAME} ...", status: 'PENDING', targetUrl: "${env.JENKINS_URL}search/?q=${env.ITS_PIPELINE.replaceAll('/','+')}")
+            build(job: env.ITS_PIPELINE, propagate: false, wait: true,
+                  parameters: [ string(name: 'INTEGRATION_TEST', value: 'Php'),
+                                string(name: 'BUILD_OPTS', value: "--php-agent-version ${env.GIT_BASE_COMMIT} --php-agent-repo ${env.CHANGE_FORK?.trim() ?: 'elastic'}/${env.REPO} --opbeans-php-agent-branch ${env.GIT_BASE_COMMIT}"),
```
Comment (v1v) [inline .ci/Jenkinsfile:290]:
```suggestion
                                string(name: 'BUILD_OPTS', value: "--php-agent-version ${env.GIT_BASE_COMMIT} --php-agent-repo ${env.CHANGE_FORK?.trim() ?: 'elastic'}/${env.REPO}"),
```

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -275,6 +277,22 @@ pipeline {
...
+        stage('Integration Tests') {
+          agent none
+          when {
+            beforeAgent true
+            changeRequest()
+          }
+          steps {
+            githubNotify(context: "${env.GITHUB_CHECK_ITS_NAME}", description: "${env.GITHUB_CHECK_ITS_NAME} ...", status: 'PENDING', targetUrl: "${env.JENKINS_URL}search/?q=${env.ITS_PIPELINE.replaceAll('/','+')}")
+            build(job: env.ITS_PIPELINE, propagate: false, wait: true,
+                  parameters: [ string(name: 'INTEGRATION_TEST', value: 'Php'),
```
Comment (v1v) [inline .ci/Jenkinsfile:289]:
```suggestion
                  parameters: [ string(name: 'INTEGRATION_TEST', value: 'PHP'),
```

**Comment by v1v** [comment]
jenkins run the tests please

**Review by SergeyKleyman — APPROVED** [review]

File: .ci/Jenkinsfile [hunk 3]
```diff
@@ -152,6 +154,22 @@ pipeline {
...
         }
+        stage('Integration Tests') {
+          agent none
+          when {
+            beforeAgent true
+            changeRequest()
+          }
+          steps {
+            githubNotify(context: "${env.GITHUB_CHECK_ITS_NAME}", description: "${env.GITHUB_CHECK_ITS_NAME} ...", status: 'PENDING', targetUrl: "${env.JENKINS_URL}search/?q=${env.ITS_PIPELINE.replaceAll('/','+')}")
+            build(job: env.ITS_PIPELINE, propagate: false, wait: true,
```
Comment (v1v) [inline .ci/Jenkinsfile:165]:
```suggestion
            build(job: env.ITS_PIPELINE, propagate: false, wait: false,
```

File: .ci/Jenkinsfile [hunk 4]
```diff
@@ -152,6 +154,22 @@ pipeline {
...
+          agent none
+          when {
+            beforeAgent true
+            changeRequest()
+          }
+          steps {
+            githubNotify(context: "${env.GITHUB_CHECK_ITS_NAME}", description: "${env.GITHUB_CHECK_ITS_NAME} ...", status: 'PENDING', targetUrl: "${env.JENKINS_URL}search/?q=${env.ITS_PIPELINE.replaceAll('/','+')}")
+            build(job: env.ITS_PIPELINE, propagate: false, wait: false,
+                  parameters: [ string(name: 'INTEGRATION_TEST', value: 'PHP'),
+                                string(name: 'BUILD_OPTS', value: "--php-agent-version ${env.GIT_BASE_COMMIT} --php-agent-repo ${env.CHANGE_FORK?.trim() ?: 'elastic/' + env.REPO}"),
```
Comment (v1v) [inline .ci/Jenkinsfile:167]: `CHANGE_FORK` is now based on `org/repo` format, otherwise

image

Comment (v1v) [inline .ci/Jenkinsfile:167]:
```suggestion
                                string(name: 'BUILD_OPTS', value: "--php-agent-version ${env.GIT_BASE_COMMIT}"),
```

**Pull request merged**